sound/soc/sof/intel/hda-trace.c

```diff
@@ -51,6 +51,15 @@ int hda_dsp_trace_init(struct snd_sof_dev *sdev, u32 *stream_tag)
 		return -ENODEV;
 	}
 
+	/*
+	 * The recommended HDA HW programming sequence requires host DMA and
+	 * link DMA with the same IDs to be coupled before setting the
+	 * DMA format. Therefore, even though trace utilizes only host DMA
+	 * in decoupled mode, the link DMA channel with the same ID must be
+	 * reserved to avoid it being allocated to another stream.
+	 */
+	hda->dtrace_stream->link_locked = 1;
```

Comment:
question: this will have no effect when the SOF_HDA option is not selected, e.g. NOCODEC?